changelog.d/12513.feature

```diff
@@ -0,0 +1 @@
+Insert Measure blocks throughout SpamChecker to collect metrics.
```

Comment:
I'd just be tempted to make this a bit more comprehensible to an administrator (I don't know if it's clear what a Measure block is to anyone except Synapse devs)..

```suggestion
Measure the time taken in spam checker callbacks and expose them as per-block metrics in Prometheus.
```

I'm no expert but I think that assumes a bit less Synapse code knowledge?

Reply:
i went a bit freestyle, but does the new version look OK?

Reply:
Ya, works for me!